### PR TITLE
[WIP][FEATURE][GAWE-74] - create edit profile page

### PR DESCRIPTION
**Pull request by azkifairuz**
#### Description

> edit profile page

#### Ticket Reference(s)

[GAWE-74](https://bluehabit.atlassian.net/browse/GAWE-74)

##### UI Changes

>
gambar
gambar

##### Coverage

> Test coverage screenshot.

N/A

#### Note(s)

> Masih belum berfungsi, hanya screen.

[GAWE-74]: https://bluehabit.atlassian.net/browse/GAWE-74?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/editProfile/EditProfileScreen.kt [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+import com.bluehabit.core.ui.components.button.ButtonPrimary
+import com.bluehabit.core.ui.components.input.InputTextPrimary
+import com.bluehabit.core.ui.components.sheet.EditPhotoBottomSheet
+import com.bluehabit.core.ui.theme.Gray900
+import com.bluehabit.eureka.data.R
+import kotlinx.coroutines.launch
+
+@Composable
+fun EditProfileScreen(
+    state: EditProfileState = EditProfileState(),
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/editProfile/EditProfileScreen.kt:67]: Ini ga perlu ya..di contract udah di provide, actionnya juga

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/editProfile/EditProfileScreen.kt [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    onNewNameChange: (String) -> Unit = {},
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/editProfile/EditProfileScreen.kt:68]: Ini gaperlu

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/editProfile/EditProfileScreen.kt [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    val context = LocalContext.current
+    fun isNameValid(): Boolean {
+        return state.newName.isNotEmpty()
+    }
+
+    val profilePicture = rememberAsyncImagePainter(
+        model = ImageRequest
+            .Builder(context)
+            .data(state.imageUrl.ifEmpty {
+                R.drawable.dummy_avatar_1
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/editProfile/EditProfileScreen.kt:85]: Ga perlu di kasih ini lgsg aja` .orEmpty()` dan ketika error dia bakal manggil di builder error

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/editProfile/EditProfileScreen.kt [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+            })
+            .placeholder(R.drawable.dummy_avatar_1)
+            .error(
+                R.drawable.dummy_avatar_1
+            )
+            .build()
+    )
+    val scope = rememberCoroutineScope()
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/editProfile/EditProfileScreen.kt:93]: Ga perlu bikin coroutine scope ya...udah di provide sama contarct langsung aja makenya `launch{}`

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/editProfile/EditProfileScreen.kt [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+                    navigationIcon = {
+                        IconButton(onClick = { /*TODO*/ }) {
+                            Icon(
+                                painter = painterResource(id = com.bluehabit.core.ui.R.drawable.ic_arrow_back),
+                                contentDescription = stringResource(id = string.description_icon_arrow)
+                            )
+                        }
+                    },
+                    actions = {
+                        IconButton(onClick = {  }) {
```
Comment (triandamai) [inline feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/editProfile/EditProfileScreen.kt:127]: Kayaknya ga dikadih button jg bisa deh

**Review by triandamai — CHANGES_REQUESTED** [review]

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/editProfile/EditProfileScreen.kt (hunk 2 above)
Comment (azkifairuz) [reply feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/editProfile/EditProfileScreen.kt:68]: buat onChangenya

File: feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/editProfile/EditProfileScreen.kt (hunk 5 above)
Comment (azkifairuz) [reply feature/feature-authentication/src/main/java/com/bluehabit/eureka/feature/authentication/editProfile/EditProfileScreen.kt:127]: pas g dikasih button g ngefek
gambar

**Review by triandamai — APPROVED** [review]

**Pull request merged**